pkg/util/util.go

```diff
@@ -0,0 +1,19 @@
+package util
+
+import (
+	"fmt"
+
+	"github.com/docker/engine-api/types/container"
+)
+
+// SafeForLoggingContainerConfig returns a string version of the container.Config object
+// with sensitive information (proxy environment variables containing credentials)
+// redacted.
+func SafeForLoggingContainerConfig(config *container.Config) string {
+	origEnv := config.Env
+	strippedEnv := StripProxyCredentials(origEnv)
+	config.Env = strippedEnv
+	s := fmt.Sprintf("%+v", config)
```

Comment:
FWIW I think I'd envisaged returning a temporary copy and having the caller decide what they'd Sprintf, but this works too :)